### PR TITLE
use github actions

### PR DESCRIPTION
**Pull request by UziTech**
This will allow github actions to run tests and eslint on every PR and merge to master.

In order to see github actions run on this PR it has to be a branch on https://github.com/ldapjs/asn1

### CODE REVIEW

**Comment by UziTech** [comment]
I commented out the Release with semantic release because it would need a token from npm to pusblish stored in a secret called NPM_TOKEN but we can enable that later if we want.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+
+jobs:
+
+  Test:
+    if: "!contains(github.event.head_commit.message, '[skip ci]')"
```
Comment (jsumners) [inline .github/workflows/main.yml:11]: I think I'll have to steal this for some other workflows 👍

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    strategy:
+      matrix:
+        node_version: ['lts/*', node]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+      - name: Install Node
+        uses: dcodeIO/setup-node-nvm@master
```
Comment (jsumners) [inline .github/workflows/main.yml:20]: Why not use the GitHub standard Node setup action?

File: package.json [hunk 3]
```diff
@@ -25,7 +25,8 @@
     "eslint-plugin-joyent": "~1.3.0"
   },
   "scripts": {
-    "test": "./node_modules/.bin/tape ./test/ber/*.test.js"
+    "test": "./node_modules/.bin/tape ./test/ber/*.test.js",
```
Comment (jsumners) [inline package.json]:
```suggestion
    "test": "tape ./test/ber/*.test.js",
```

**Comment by jsumners** [comment]
> I commented out the Release with semantic release because it would need a token from npm to pusblish stored in a secret called NPM_TOKEN but we can enable that later if we want.

Automating publishes to NPM is tricky if you have 2FA required enabled.

File: .github/workflows/main.yml (hunk 2 above)
Comment (UziTech) [reply .github/workflows/main.yml:20]: actions/setup-node does not support aliases like `lts/*` or `latest` https://github.com/actions/setup-node/issues/26

**Comment by UziTech** [comment]
> Automating publishes to NPM is tricky if you have 2FA required enabled.

True. I have 2FA set to `auth-only`. There are some security concerns but I feel like it is worth the risk to allow publishing without 2FA since my password and the tokens are sufficiently long enough to be unguessable. NPM sends an email every time a package is published so it wouldn't take long to figure out if someone published something malicious.

File: .github/workflows/main.yml (hunk 2 above)
Comment (jsumners) [reply .github/workflows/main.yml:20]: I'd rather be explicit anyway and list out `[10, 12, 14]`.

**Comment by jsumners** [comment]
> so it wouldn't take long to figure out if someone published something malicious.

Still too late.

**Comment by UziTech** [comment]
> Still too late.

Perhaps. There is always a trade off between security and convenience.

**Comment by UziTech** [comment]
Did you want me to remove the `Release` section or leave it commented?

**Comment by jsumners** [comment]
The release cycle here is going to be so low, that automation is going to add to the maintenance burden, not lessen it.

**Comment by jsumners** [comment]
> Did you want me to remove the `Release` section or leave it commented?

Let's just remove it.

**Review by jsumners — APPROVED** [review]
LGTM

**Pull request merged**